modules/rostests/kmtests/hal/HalPortIo.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * PROJECT:     ReactOS Kernel-Mode Tests
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Tests for string I/O intrinsic functions
+ * COPYRIGHT:   Copyright 2025 Dmitry Borisov <di.sean@protonmail.com>
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include <kmt_test.h>
+
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS ********************************************************************/
+
+/* Trigger a bug in the CORE-20078 */
+#ifdef _MSC_VER
+#pragma optimize("gst", on)
+#pragma auto_inline(on)
+#else
+#pragma GCC optimize("O3")
+#endif
+
+/* Used to isolate the effects of intrinsics from each other */
+typedef struct _TEST_CONTEXT
+{
+    PVOID OldBuffer;
+    UCHAR Pad[78];
+    ULONG Port;
+    ULONG Size;
+    PVOID Buffer;
+} TEST_CONTEXT, *PTEST_CONTEXT;
+
+/* FUNCTIONS ******************************************************************/
+
+static
+/* Isolate the effects of intrinsics from each other */
+DECLSPEC_NOINLINE
+VOID
+TestWriteStringUshort(
+    _In_ PTEST_CONTEXT Context)
+{
+    __outwordstring(Context->Port, Context->Buffer, Context->Size / sizeof(USHORT));
+
+    /* Test for CORE-20078 */
```

Comment:
Trying to be more explicit about what is happening, something like:
```suggestion
    /* Test for CORE-20078: compare locally optimized Buffer storage to OldBuffer value in struct */
```